src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -629,8 +629,27 @@ private bool StartWithCreateProcess(ProcessStartInfo startInfo)
             return ret;
         }
 
-        private static Encoding GetEncoding(int codePage)
+        private static Encoding GetConsoleEncodingOrThrow(ConsoleStreamDirection encodingType)
         {
+            int codePage = (encodingType == ConsoleStreamDirection.Input) ?
+                    (int)Interop.Kernel32.GetConsoleCP() :
+                    (int)Interop.Kernel32.GetConsoleOutputCP();
+
+            int error = Marshal.GetLastWin32Error();
+            if (error == Interop.Errors.ERROR_INVALID_HANDLE)
+            {
+                throw new PlatformNotSupportedException(SR.ConsoleEncodingNotSupported);
```

Comment:
I thought we did not want this specific behavior. I can see how this could work when passing binary data on the stdin but writing text with random encoding would be unsafe this way. cc: @tarekgh